extensions/amp-social-share/0.1/amp-social-share.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {getSocialConfig} from './amp-social-share-config';
+import {addParamsToUrl} from '../../../src/url.js';
+import {closest} from '../../../src/dom.js';
+
+const DEFAULT_WIDTH = 60;
+const DEFAULT_HEIGHT = 44;
+
+const CONTAINER_TAG = 'span';
+const LINK_TAG = 'a';
+
+class AmpSocialShare extends AMP.BaseElement {
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return isLayoutSizeDefined(layout);
+  }
+
+  /** @override */
+  buildCallback() {
+    this.config_ = this.getElementConfig(this.element);
+    AMP.assert(this.config_, 'The <script> config is required for ' +
+      '<amp-social-share> %s');
+
+    this.type_ = AMP.assert(this.element.getAttribute('type'),
```

Comment:
please annotate the fields/properties. this one should be `/** @private @const {string} */`
see https://developers.google.com/closure/compiler/docs/js-for-compiler#types for further info. (please don't hesitate to ask as well)